### ISSUE TITLE
Tranche deposits during an outstanding recoverable loss capture the insurance recovery from incumbent LPs (generalizes #139)

### ISSUE THREAD

**Issue by 0x-SquidSol** [dcccrypto/percolator-stake#145]
## Summary

When a flushed insurance loss is still **outstanding-but-recoverable** (`total_flushed > total_returned`), tranche deposits are priced at the **loss-depressed** sub-pool value (`effective_junior_balance()` / `senior_balance()`). An unprivileged user can deposit into the tranche during that window, then — once the loss is recovered via `ReturnInsurance` — redeem at the restored price, **capturing the recovery that belongs to the incumbent LPs who actually bore the loss**. It is a conservation-exact, unprivileged value transfer, reachable in the **normal lifecycle** (deposit while the market is live/pre-resolution; no special trick).

This is the **general case** of the previously-reported insurance-return snipe (#139): #139 required the post-resolution window (a stale `market_resolved` flag under permissionless resolution); this one needs no such trick — the live pre-resolution deposit window is always open while a loss is outstanding. A fix that gates tranche deposits during an outstanding loss closes **both**.

## Affected code

- `src/state.rs` — `effective_junior_balance()` (loss-adjusts the junior sub-pool by `net_loss = total_flushed - total_returned`) and `senior_balance()`.
- `src/processor.rs` — `process_deposit_junior` / senior path of `process_deposit` price against the loss-adjusted sub-pool balance; `process_flush_to_insurance` creates the loss; `process_return_insurance` credits `total_returned` (the recovery). Junior/senior deposits are **not** gated on an outstanding loss.

## Mechanism (junior tranche; senior is the same once a loss spills past junior)

1. Incumbent junior **Alice** deposits 1000 → `junior_balance=1000, junior_total_lp=1000`.
2. Admin flushes 800 to insurance (recoverable loss). `effective_junior_balance = 1000 - 800 = 200` — Alice's tranche is marked to 200.
3. **Eve** deposits junior 200 during the live loss window, priced at the depressed 200: `calc_junior_lp_for_deposit(1000, 200, 200) = 1000` LP. Eve now holds 1000 junior LP for a 200 deposit, equal to Alice's 1000 LP.
4. Market resolves; admin returns the 800 (`ReturnInsurance`) → `total_returned=800`, `net_loss=0`, `effective_junior_balance = 1200` across `junior_total_lp = 2000`.
5. Eve withdraws 1000 LP → `1000 * 1200 / 2000 = 600` (deposited 200, **+400**).
6. Alice withdraws 1000 LP → 600 (deposited 1000; should have recovered her full 1000 once the loss was returned — **−400**).

Eve **+400**, Alice **−400**: a clean transfer. Eve's downside is bounded (she paid fair depressed value; if the loss is never recovered she breaks even), so the strategy is net-positive in expectation whenever a return is plausible — an asymmetric "buy the recoverable dip" on capital she never put at risk. Deeper losses make the dilution cheaper (depositing against `effective ≈ small` mints a large LP share for little).

## Impact

Direct, unprivileged transfer from the loss-absorbing incumbent tranche LPs to a just-in-time depositor. Scales with loss depth and attacker capital. The withdrawal cooldown only delays the attacker's exit (the flush→resolve→return cycle spans an epoch); it does not prevent the dilution, which is locked in at deposit time.

## Reachability

Any unprivileged user, on a tranche-enabled pool, given (i) an admin flush that partially hits the sub-pool and (ii) a later `ReturnInsurance`. Both the flush and the insurance balance are on-chain observable, so the recovery is predictable. No privilege is needed for the profitable step.

## Proof of concept

State/math-level test in the repo convention (no runtime), exercising the exact production functions. Add as `tests/poc_tranche_recovery_snipe.rs`:

```rust
use bytemuck::Zeroable;
use percolator_stake::math::{calc_junior_collateral_for_withdraw, calc_junior_lp_for_deposit};
use percolator_stake::state::StakePool;

fn pool() -> StakePool {
    let mut p = StakePool::zeroed();
    p.is_initialized = 1;
    p.set_discriminator();
    p.set_tranche_enabled(true);
    p // pool_mode 0 (insurance LP)
}

#[test]
fn tranche_recovery_snipe_transfers_from_loss_bearer() {
    let mut pool = pool();

    // Incumbent junior Alice deposits 1000.
    pool.set_junior_balance(1000);
    pool.set_junior_total_lp(1000);
    pool.total_deposited = 1000;
    pool.total_lp_supply = 1000;

    // Admin flushes 800 (recoverable loss). Junior marked to 200.
    pool.total_flushed = 800;
    let eff = pool.effective_junior_balance();
    assert_eq!(eff, 200);

    // Eve deposits junior 200 during the LIVE (pre-resolution) loss window.
    let eve_lp = calc_junior_lp_for_deposit(pool.junior_total_lp(), eff, 200).unwrap();
    assert_eq!(eve_lp, 1000);
    pool.set_junior_balance(pool.junior_balance() + 200);
    pool.set_junior_total_lp(pool.junior_total_lp() + eve_lp);
    pool.total_deposited += 200;
    pool.total_lp_supply += eve_lp;

    // Market resolves; admin returns the 800 insurance.
    pool.total_returned = 800;
    assert_eq!(pool.effective_junior_balance(), 1200);

    // Eve withdraws: 1000 * 1200 / 2000 = 600 for a 200 deposit -> +400.
    let eve_back =
        calc_junior_collateral_for_withdraw(pool.junior_total_lp(), pool.effective_junior_balance(), eve_lp).unwrap();
    assert_eq!(eve_back, 600);
    pool.set_junior_balance(pool.junior_balance() - eve_back);
    pool.set_junior_total_lp(pool.junior_total_lp() - eve_lp);

    // Alice withdraws: 600 — should have recovered her full 1000 once the loss returned.
    let alice_back =
        calc_junior_collateral_for_withdraw(pool.junior_total_lp(), pool.effective_junior_balance(), 1000).unwrap();
    assert_eq!(alice_back, 600);

    assert_eq!(eve_back - 200, 400, "Eve net +400");
    assert_eq!(1000 - alice_back, 400, "Alice net -400 — her recovery, taken by Eve");
}
```

(Verified out-of-band against the v17 production functions: Eve +400, Alice −400.)

## Possible fixes (design decision)

The loss-adjusted deposit pricing is **intentional** — it avoids *overcharging* a new depositor relative to the current mark-to-market value. The asymmetry is specifically that an already-flushed (pre-deposit) loss is **recoverable**, so a JIT depositor shares a recovery they never had at risk. Options:

1. **Gate tranche deposits while a loss is outstanding** (`total_flushed > total_returned`): reject `process_deposit_junior` / the senior deposit path during an unreturned flush. Simple; closes this **and** #139. Trade-off: blocks legitimate new tranche capital during a loss window.
2. **Per-LP / per-cohort loss attribution**: track which LPs bore the flush so the recovery returns to them. Correct but materially more complex.
3. **Accept as a documented characteristic** of tranche LPs (entering during a loss is a risk position; the recovery upside is the reward) — but then #139's post-resolution variant must still be closed separately.

Recommendation: option 1 (it also subsumes #139), unless the protocol wants to keep deposits open during losses, in which case option 2.

**Comment by dcccrypto** [comment]
Verified (workflow + refute) — **reduces to the already-closed #139, by-design**. The post-flush/pre-return dilution math is real and reproduces (a depositor priced on the temporarily-depressed NAV captures some recovery), BUT the profitable step depends entirely on a **discretionary, admin-funded** `ReturnInsurance` (the admin transfers their OWN tokens to raise NAV — there's no free/permissionless NAV bump). Same ruling as #139: an admin choosing to inject their own capital and thereby benefit a window of depositors is an admin economic decision, not an unprivileged extraction. No code change required for security. Closing.
